src/System.Net.Primitives/src/System/Net/IPAddressParser.cs

```diff
@@ -54,24 +62,76 @@ internal static unsafe string IPv4AddressToString(uint address)
         {
             const int MaxLength = 15;
             char* addressString = stackalloc char[MaxLength];
-            int offset = MaxLength;
 
-            FormatIPv4AddressNumber((int)((address >> 24) & 0xFF), addressString, ref offset);
-            addressString[--offset] = '.';
-            FormatIPv4AddressNumber((int)((address >> 16) & 0xFF), addressString, ref offset);
-            addressString[--offset] = '.';
-            FormatIPv4AddressNumber((int)((address >> 8) & 0xFF), addressString, ref offset);
-            addressString[--offset] = '.';
+            int charsWritten = IPv4AddressToStringHelper(address, new Span<char>((void*)addressString, MaxLength));
```

Comment:
You shouldn't need to explicitly cast: a `char*` is implicitly convertible to a `void*`.